code_challenges/code_wars/maximum_subarray_sum.py

```diff
@@ -0,0 +1,37 @@
+def max_sequence(sequence):
+    total = 0
+    for number in sequence:
+        if number > 0:
+            total += 1
+    if total == 0:
+        return total
+    else:
+        for i, number in enumerate(sequence):
+            j = len(sequence)
+            while j > i:
+                if sequence[j-1] > 0:
+                    if sum(sequence[i:j]) > total:
+                        total = sum(sequence[i:j])
+                        j -= 1 
+                    else:
+                        j -= 1
+                else:
+                    j -= 1        
```

Comment:
In all three case you minus equal 1.  Only write it once outside the if block as the last thing in the while block.